unused-variable emitted instead of unused-import

<!--
  Hi there! Thank you for discovering and submitting an issue.

  Before you submit this, make sure that the issue doesn't already exist
  or if it is not closed.

  Is your issue fixed on the preview release?: pip install pylint astroid --pre -U

--> No

### Steps to reproduce
1. cat sample6.py
```
def foobar():
    from functools import wraps
```
2. pylint sample6.py

### Current behavior
************* Module sample6
sample6.py:2:4: W0612: Unused variable 'wraps' (unused-variable)

------------------------------------------------------------------
Your code has been rated at 5.00/10 (previous run: 0.00/10, +5.00)


### Expected behavior
Instead of unused-variable, unused-import should be raised.

The behaviour is correct when import statement is at top. e.g
```
from functools import wraps

def foobar():
    pass

```
pylint sample6.py

************* Module sample6
sample6.py:1:0: W0611: Unused wraps imported from functools (unused-import)

------------------------------------------------------------------
Your code has been rated at 6.67/10 (previous run: 5.00/10, +1.67)


### pylint --version output
pylint 2.2.0
astroid 2.0.1
Python 3.6.4 (default, Jan  7 2018, 15:53:53)
[GCC 6.4.0]

--- a/ChangeLog
+++ b/ChangeLog
@@ -6,6 +6,10 @@ What's New in Pylint 2.2?
 =========================
 
 Release date: TBA
+   * ``pylint`` used to emit a ``unused-variable`` error if unused import was found in the function. Now instead of
+     ``unused-variable``, ``unused-import`` is emitted.
+
+     Close #2421
 
    * Handle asyncio.coroutine when looking for ``not-an-iterable`` check.
 

--- a/doc/whatsnew/2.2.rst
+++ b/doc/whatsnew/2.2.rst
@@ -36,3 +36,8 @@ Other Changes
   with ``asyncio.coroutine``. This is no longer the case as we handle coroutines explicitly.
 
   Close #996
+
+* ``pylint`` used to emit a ``unused-variable`` error if unused import was found in the function. Now instead of
+  ``unused-variable``, ``unused-import`` is emitted.
+
+  Close #2421

--- a/pylint/checkers/variables.py
+++ b/pylint/checkers/variables.py
@@ -262,7 +262,7 @@ def _assigned_locally(name_node):
 
 
 def _has_locals_call_after_node(stmt, scope):
-    skip_nodes = (astroid.FunctionDef, astroid.ClassDef)
+    skip_nodes = (astroid.FunctionDef, astroid.ClassDef, astroid.Import, astroid.ImportFrom)
     for call in scope.nodes_of_class(astroid.Call, skip_klass=skip_nodes):
         inferred = utils.safe_infer(call.func)
         if utils.is_builtin_object(inferred) and getattr(inferred, 'name', None) == 'locals':
@@ -793,9 +793,7 @@ class VariablesChecker(BaseChecker):
 
         # Ignore names imported by the global statement.
         # FIXME: should only ignore them if it's assigned latter
-        if isinstance(stmt, astroid.Global):
-            return
-        if isinstance(stmt, (astroid.Import, astroid.ImportFrom)):
+        if isinstance(stmt, (astroid.Global, astroid.Import, astroid.ImportFrom)):
             # Detect imports, assigned to global statements.
             if global_names and _import_name_is_global(stmt, global_names):
                 return
@@ -837,7 +835,8 @@ class VariablesChecker(BaseChecker):
                 if name in nonlocal_names:
                     return
 
-            if isinstance(stmt, astroid.Import):
+
+            if isinstance(stmt, (astroid.Import, astroid.ImportFrom)):
                 # Need the complete name, which we don't have in .locals.
                 qname, asname = stmt.names[0]
                 name = asname or qname
@@ -845,7 +844,22 @@ class VariablesChecker(BaseChecker):
             if _has_locals_call_after_node(stmt, node.scope()):
                 message_name = 'possibly-unused-variable'
             else:
-                message_name = 'unused-variable'
+                if isinstance(stmt, astroid.Import):
+                    if asname is not None:
+                        msg = "%s imported as %s" % (qname, asname)
+                    else:
+                        msg = "import %s" % name
+                    self.add_message('unused-import', args=msg, node=stmt)
+                    return
+                elif isinstance(stmt, astroid.ImportFrom):
+                    if asname is not None:
+                        msg = "%s imported from %s as %s" % (qname, stmt.modname, asname)
+                    else:
+                        msg = "%s imported from %s" % (name, stmt.modname)
+                    self.add_message('unused-import', args=msg, node=stmt)
+                    return
+                else:
+                    message_name = 'unused-variable'
             self.add_message(message_name, args=name, node=stmt)
 
     def leave_functiondef(self, node):

--- a/pylint/test/functional/unused_variable.py
+++ b/pylint/test/functional/unused_variable.py
@@ -1,11 +1,11 @@
 # pylint: disable=missing-docstring, invalid-name, too-few-public-methods, no-self-use, useless-object-inheritance
 
 def test_regression_737():
-    import xml # [unused-variable]
+    import xml # [unused-import]
 
 def test_regression_923():
-    import unittest.case  # [unused-variable]
-    import xml as sql # [unused-variable]
+    import unittest.case  # [unused-import]
+    import xml as sql # [unused-import]
 
 def test_unused_with_prepended_underscore():
     _foo = 42
@@ -48,3 +48,8 @@ def locals_does_not_account_for_subscopes():
     def some_other_scope():
         return locals()
     return some_other_scope
+
+
+def unused_import_from():
+    from functools import wraps as abc # [unused-import]
+    from collections import namedtuple # [unused-import]

--- a/pylint/test/functional/unused_variable.txt
+++ b/pylint/test/functional/unused_variable.txt
@@ -1,6 +1,6 @@
-unused-variable:4:test_regression_737:Unused variable 'xml'
-unused-variable:7:test_regression_923:Unused variable 'unittest.case'
-unused-variable:8:test_regression_923:Unused variable 'sql'
+unused-import:4:test_regression_737:Unused import xml
+unused-import:7:test_regression_923:Unused import unittest.case
+unused-import:8:test_regression_923:Unused xml imported as sql
 unused-variable:15:test_unused_with_prepended_underscore:Unused variable '_a_'
 unused-variable:16:test_unused_with_prepended_underscore:Unused variable '__a__'
 unused-variable:20:test_local_field_prefixed_with_unused_or_ignored:Unused variable 'flagged_local_field'
@@ -8,3 +8,5 @@ unused-variable:28:HasUnusedDunderClass.test:Unused variable '__class__'
 possibly-unused-variable:35:locals_example_defined_before:Possibly unused variable 'value'
 unused-variable:41:locals_example_defined_after:Unused variable 'value'
 unused-variable:46:locals_does_not_account_for_subscopes:Unused variable 'value'
+unused-import:54:unused_import_from:Unused wraps imported from functools as abc
+unused-import:55:unused_import_from:Unused namedtuple imported from collections

--- a/pylint/test/messages/func_w0612.txt
+++ b/pylint/test/messages/func_w0612.txt
@@ -1,6 +1,6 @@
 W:  9:function: Unused variable 'aaaa'
 W: 28:test_global: Using the global statement
-W: 34:test_global: Unused variable 'platform'
-W: 35:test_global: Unused variable 'VERSION'
-W: 36:test_global: Unused variable 'this'
-W: 37:test_global: Unused variable 'RE'
+W: 34:test_global: Unused platform imported from sys
+W: 35:test_global: Unused version imported from sys as VERSION
+W: 36:test_global: Unused import this
+W: 37:test_global: Unused re imported as RE

